frontend/src/lib/Segment.ts

```diff
@@ -0,0 +1,59 @@
+declare global {
+  interface Window {
+    analytics: any
+  }
+}
+
+export const segmentLauncher = (): void => {
+  const analytics = (window.analytics = window.analytics || [])
```

Comment:
It looks like this is the Prettier formatted version of the minified code. It's much easier to read, which is a plus, but still has some weird quirks of minified code (like !0 instead of true).

Let's replace this with the non-minified version here, complete with comments: https://segment.com/docs/connections/sources/catalog/libraries/website/javascript/quickstart/
(and mention in a JSDoc where we got this code, so future programmers can figure out if we need to make any changes.)

Reply:
Added the non-minified version and made minor changes to pass the linting (for eg. var changed to const or let)
Added JSDoc and some documentation with references